pkg/client/record/event_test.go

```diff
@@ -540,6 +578,27 @@ func TestEventfNoNamespace(t *testing.T) {
 			updateEvent <- event
 			return event, nil
 		},
+		OnPatch: func(event *api.Event, patch []byte) (*api.Event, error) {
+			cachedEvent, found := testCache[getEventKey(event)]
+			if !found {
```

Comment:
Yeah, I have to copy the code because I need to use the free variable testCache and updateEvent etc. I'll discuss with Saad to see if there is a better way.